### ISSUE TITLE
Make IntelliJ inspection "Unused declaration" a error in the whole codebase

### ISSUE THREAD

**Issue by leventov** [apache/incubator-druid#6860]
Currently it's considered an error only in a few core modules.

A follow-up of #5461.

**Comment by gianm** [comment]
Not specifically related to this issue but more of a general thing. Should we be recommending IDEA for Druid developers? I see a lot of people using Eclipse and I feel that the more IntelliJ-specific stuff we add, the less friendly the developer experience is to people using Eclipse. We might want to recommend they bite the bullet and switch to IDEA. (Or alternatively: deal with the fact that some static analysis won't run for them until they raise a PR, and they'll need to check CI output.)

**Comment by leventov** [comment]
IntelliJ users don't really become aware of inspection violations until a PR is created too. A full analysis takes about an hour in TeamCity.

**Comment by gianm** [comment]
Anecdotally I've seen a few of these inspections show up as warnings and errors in the IntelliJ IDE. I have definitely seen it flag unused declarations (although I think they show up as warnings rather than errors).

**Comment by leventov** [comment]
Of course, IntelliJ shows a lot of warnings, much more than we enforce (yet). I believe Eclipse shows warnings too, many of them should intersect with what IntelliJ shows. Most of those warnings point to actual (albeit often minor) problems with code. Ideally developers should proactively fix them as much as possible, not waiting until they are enforced in TeamCity.
